### PR TITLE
feat: replace verilog parser + use upstream iso forked version

### PR DESCRIPTION
**Pull request by henrykvdb**
Change:
* Replace @zhangwwpeng 's personal fork of systemverilog parser by upstream version, since now the compiled parser.c file is also included there. I kept him as the maintainer for this parser however. If he is still planning to contribute to this project I encourage him to submit the changes upstream as it is still actively maintained.
* Replace verilog parser by this parser also, see https://github.com/nvim-treesitter/nvim-treesitter/pull/6775 for context on this change

To make this change work also the textobjects need to be updated, so preferably this PR is accepted together with
https://github.com/nvim-treesitter/nvim-treesitter-textobjects/pull/678

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -451,7 +450,7 @@ We are looking for maintainers to add more parsers and to write query files for
 - [x] [v](https://github.com/vlang/v-analyzer) (maintained by @kkharji, @amaanq)
 - [x] [vala](https://github.com/vala-lang/tree-sitter-vala) (maintained by @Prince781)
 - [x] [vento](https://github.com/ventojs/tree-sitter-vento) (maintained by @wrapperup, @oscarotero)
-- [x] [verilog](https://github.com/tree-sitter/tree-sitter-verilog) (maintained by @zegervdv)
+- [x] [verilog](https://github.com/tree-sitter/tree-sitter-verilog) (maintained by @zhangwwpeng)
```
Comment (clason) [inline README.md:453]:
```suggestion
- [x] [verilog](https://github.com/gmlarumbe/tree-sitter-verilog) (maintained by @zhangwwpeng)
```

File: queries/verilog/injections.scm [hunk 2]
```diff
@@ -1,2 +1,5 @@
 ((comment) @injection.content
   (#set! injection.language "comment"))
+
+((macro_text) @injection.content
+  (#set! injection.language "systemverilog"))
```
Comment (clason) [inline queries/verilog/injections.scm:5]:
```suggestion
  (#set! injection.language "verilog"))
```
(There's also `@injection.self`)

File: queries/verilog/locals.scm
Comment (clason) [inline queries/verilog/locals.scm:1]: Why was this file deleted?

Comment (zhangwwpeng) [reply queries/verilog/locals.scm:1]: The locals.scm in System Verilog has not been created yet, and I don't know what it is used for. If it is necessary, I am willing to contribute to it

**Comment by clason** [comment]
@henrykvdb friendly ping

File: README.md (hunk 1 above)
Comment (clason) [reply README.md:453]: And similarly in the README.

**Comment by clason** [comment]
If you're not making manual changes to the `verilog` queries that are not part of the `systemverilog` queries, using `git mv` would lead to a cleaner history and be much prefered.

**Comment by henrykvdb** [comment]
> If you're not making manual changes to the `verilog` queries that are not part of the `systemverilog` queries, using `git mv` would lead to a cleaner history and be much preferred.

Fixed this and the other things, thanks for the review! Sorry for the wait, I am in the middle of a move currently.

As for the locals.scm that seemed to be missing from the new version.
@zhangwwpeng are you looking further into it in the end? If not I can also do it, but will most likely only find time around November.

**Comment by clason** [comment]
@henrykvdb It looks like `git mv` didn't make it through. Can you maybe open a clean PR (from the tip of master) and make sure you `git mv` the systemverilog queries _first_ and then make changes?

**Comment by henrykvdb** [comment]
Screenshot_20241015-091812

Hi @clason if you select the appropriate commit where the move happened it shows exactly as expected.

**Comment by clason** [comment]
Yes, but the _total_ change does not reflect the move, and we'll need to squash the commits. Or did you make manual changes to the queries?

**Comment by clason** [comment]
(and needs a rebase anyway)

Note:
1. Never open a PR from your master branch.
2. Never merge master into your branch; always use `git rebase`.

Otherwise it messes up history and prevents maintainers from pushing to the PR to fix issues themselves.

**Comment by henrykvdb** [comment]
Hi I did all the steps again from scratch as you wanted and the resulting composite history looks ugly as before.
Feel free to inspect any commit individually, they all are as expected since I used the proper commands. 

first commit: git rm queries/verilog
second commit: git mv queries/systemverilog to queries/verilog
third commit: make small single line correction in queries/verilog (formally known as queries/systemverilog)
fourth commit: update all other things

**Comment by henrykvdb** [comment]
But please review again since I did it from scratch, probably best go commit per commit because otherwise the history looks weird
